testutils/consul.go

```diff
@@ -0,0 +1,34 @@
+package testutils
+
+import (
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/consul/sdk/testutil"
+)
+
+func NewTestConsulServerHTTPS(tb testing.TB, path string) *testutil.TestServer {
```

Comment:
This could probably be further refactored to eliminate the duplicate code in e2e_test.go, `newTestConsulServer()`. 

Not really related to this issue though. Maybe more a suggestion for a later cleanup sweep.

Reply:
+1! i can make a follow-up PR.

I'm not too happy with the function needing the path as a parameter to discover the certs. It's based off of the caller's test file location and not relative to this function.

Reply:
https://github.com/hashicorp/consul-terraform-sync/pull/226